src/core/server/saved_objects/types.ts

```diff
@@ -208,13 +208,65 @@ export type MutatingOperationRefreshSetting = boolean | 'wait_for';
  */
 export type SavedObjectsClientContract = Pick<SavedObjectsClient, keyof SavedObjectsClient>;
 
+/**
+ * @remarks This is only internal for now, and will only be public when we expose the registerType API
+ *
+ * @internal
+ */
+export interface SavedObjectsType {
+  name: string;
+  hidden: boolean;
+  namespaceAgnostic: boolean;
+  indexPattern?: string;
+  convertToAliasScript?: string;
+
+  mappings: SavedObjectsTypeMappingDefinition;
+  migrations?: SavedObjectMigrationMap;
```

Comment:
Could we add documentation for each of these options? Some of the names are not inherently obvious (`namespaceAgnostic`, `indexPattern`, `convertToAliasScript`). Can be done in PR that makes this publicly available to plugins.

Reply:
There is a lot of work to do on documentation, but yes, I wanted to regroup everything doc-related on the 'public API' PR (as documentation will be 80% of the work on this PR). These types are still internal ATM.

Unless of course documentation is required for proper code review (the domain is not the easiest of core), then please tell me so.

Reply:
Doing in separate PR is fine.

Reply:
Added documentation for `SavedObjectType` and underlying `migration` types. Migrations doc should probably be improved, but once again, we have the actual public API PR to do so.